utility/map_str_enhancer.go

```diff
@@ -148,7 +135,7 @@ func Add(m common.MapStr, key string, val interface{}) {
 		v := reflect.ValueOf(val)
 		switch v.Type().Kind() {
 		case reflect.Slice, reflect.Array:
-			if v.Len() == 0 {
+			if v.Len() == 0 && remove {
```

Comment:
does that mean we'll store an empty slice/array if remove is not set? is that what we want?